Artigos/O protocolo HTTP.md

```diff
@@ -0,0 +1,56 @@
+
+## O que é o Protocolo HTTP?
+
+O Protocolo HTTP é um protocolo de comunicação utilizado para transferir dados na internet. Ele opera no modelo cliente-servidor, onde um cliente (geralmente um navegador web) faz requisições a um servidor para acessar recursos, como páginas da web, imagens ou vídeos, através de URLs (Uniform Resource Locators).
+
+## Funcionamento do HTTP:
+
+O HTTP opera em um modelo de solicitação-resposta, seguindo várias etapas de comunicação:
+
+1. **Estabelecimento da Conexão:** O cliente inicia a comunicação estabelecendo uma conexão TCP/IP com o servidor na porta padrão 80 para HTTP e na porta 443 para HTTPS (HTTP Secure).
+2. **Envio da Requisição:** O cliente envia uma mensagem de requisição HTTP contendo informações como o método da requisição, a URL do recurso desejado, cabeçalhos de requisição e, opcionalmente, o corpo da requisição com dados adicionais.
+3. **Processamento da Requisição:** O servidor recebe a requisição, interpreta os dados e determina como responder. Ele executa a ação solicitada pelo cliente e prepara uma resposta apropriada.
+4. **Envio da Resposta:** O servidor envia uma mensagem de resposta HTTP de volta ao cliente contendo o código de status da resposta, cabeçalhos de resposta e, opcionalmente, o corpo da resposta com os dados solicitados pelo cliente.
+5. **Fechamento da Conexão:** Após completar a transação, a conexão TCP pode ser fechada, dependendo das configurações do servidor e do cliente.
+
+## Requisições HTTP:
+
+Existem vários tipos de métodos HTTP que podem ser utilizados em uma requisição. Alguns dos métodos mais comuns incluem:
+
+- **GET:** Solicita a representação de um recurso específico.
+- **POST:** Submete dados para serem processados por um recurso identificado pela URL.
+- **PUT:** Substitui todas as representações atuais do recurso de destino com o carregado no corpo da requisição.
+- **DELETE:** Remove o recurso especificado.
+- **PATCH:** Aplica modificações parciais a um recurso.
+- **HEAD:** Solicita apenas os cabeçalhos de resposta, sem o corpo da resposta.
+
+## Códigos de Status HTTP:
+
+Após receber uma requisição, o servidor responde com um código de status HTTP, que indica o resultado da operação. Alguns dos códigos de status mais comuns incluem:
+
+- **1xx:** Informacional - Indica que a requisição foi recebida e está sendo processada.
+- **2xx:** Sucesso - Indica que a requisição foi bem-sucedida.
+- **3xx:** Redirecionamento - Indica que é necessário tomar uma ação adicional para completar a requisição.
+- **4xx:** Erro do Cliente - Indica que ocorreu um erro na requisição do cliente.
+- **5xx:** Erro do Servidor - Indica que ocorreu um erro no servidor enquanto processava a requisição.
+
+## Exemplo Prático:
```

Comment:
Caso for considerar todo o caminho do navegador até ter a página carregada é necessário considerar mais etapas, como a tradução do domínio (DNS), e no caso do HTTPS a troca das chaves do protoloco TLS